kdc/kerberos5.c

```diff
@@ -2052,6 +2121,14 @@ _kdc_as_rep(astgs_request_t r)
 	kdc_log(r->context, config, 4, "UNKNOWN -- %s: %s", r->cname, msg);
 	krb5_free_error_message(r->context, msg);
 	ret = KRB5KDC_ERR_C_PRINCIPAL_UNKNOWN;
+	_kdc_audit_auth_status(r, HDB_AUTH_CLIENT_UNKNOWN, NULL, NULL);
+	goto out;
+    }
+    if (r->client->entry.flags.locked_out) {
+	ret = KRB5KDC_ERR_CLIENT_REVOKED;
+	kdc_log(r->context, r->config, 0,
+		"Client (%s) is locked out", r->cname);
+	_kdc_audit_auth_status(r, HDB_AUTH_CLIENT_LOCKED_OUT, NULL, NULL);
```

Comment:
This hunk looks wrong, I think the logic is from before #788 was merged as it would, if I read correctly, apply the locked_out logic to a GSS or PKINIT login. 

Either way, we should not be changing code flow in this PR, that should be in a distinct patch if needed later.

Reply:
It may be that the `_kdc_audit_auth_status()` bit is needed elsewhere, eg at the sites patched by 33fccb8bbec20b8a01263b629571404662b3a9c3